clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1239,9 +1239,9 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
   if (getLangOpts().OpenMP && CurCodeDecl)
     CGM.getOpenMPRuntime().emitFunctionProlog(*this, CurCodeDecl);
 
-  if (FD && getLangOpts().HLSL) {
+  if (getLangOpts().HLSL) {
     // Handle emitting HLSL entry functions.
-    if (FD->hasAttr<HLSLShaderAttr>()) {
+    if (FD && FD->hasAttr<HLSLShaderAttr>()) {
       CGM.getHLSLRuntime().emitEntryFunction(FD, Fn);
     }
     CGM.getHLSLRuntime().setHLSLFunctionAttributes(FD, Fn);
```

Comment:
Maybe not for this change, but a bit of me is wondering if the two CGHLSLRuntime functions emitEntryFunction and setHLSLFunctionAttributes should be somewhat combined so that we end up with something like:

```c++
if (getLangOpts().HLSL)
    CGM.getHLSLRuntime().emitFunctionProlog(FD, Fn);
```

Similar to how this is encapsulated for OpenMP above.  That way we get more of the HLSL-specific stuff in CGHLSLRuntime.cpp.

Reply:
Yeah, the `emitFunctionProlog` form is how it looked in an earlier version of a change Helena made. I think it was #102275 which introduced `setHLSLFucntionAttributes`. I decided to follow her lead, but I don't know the reasoning behind it. 

This becomes moot with the new approach I'm taking.